packages/relay/src/lib/eth.ts

```diff
@@ -1270,7 +1270,15 @@ export class EthImpl implements Eth {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     // Execute the call and get the response
     this.logger.debug(`${requestIdPrefix} Making eth_call on contract ${call.to} with gas ${gas} and call data "${call.data}" from "${call.from}" using consensus-node.`, call.to, gas, call.data, call.from);
-    
+
+    // If "From" is distinct from blank, we check is a valid account
+    if (call.from) {
+      const fromEntityType = await this.mirrorNodeClient.resolveEntityType(call.from, [constants.TYPE_ACCOUNT], requestId);
```

Comment:
nit: for a future PR we should just substitute this for a simple getAccount call

Reply:
agreed.